tests/src/Simple/Threading/Threading.cs

```diff
@@ -758,6 +789,282 @@ public static void MutexMaximumReacquireCountTest()
+        for(int i = 0; i < Environment.ProcessorCount; ++i)
+        {
+            Task.Factory.StartNew(() => {
+                CultureInfo.CurrentCulture = cultureInfo;
+                CultureInfo.CurrentUICulture = cultureInfo;
+                Thread.CurrentThread.Priority = ThreadPriority.AboveNormal;
+                if(Interlocked.Increment(ref count) == Environment.ProcessorCount)
+                {
+                    e0.Set();
+                }
+            });
+        }
+        Assert.True(e0.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds));
+        // Run the test again to make sure we can reuse the threads.
+        count = 0;
+        for(int i = 0; i < Environment.ProcessorCount; ++i)
+        {
+            Assert.Equal(expectedCultureInfo, CultureInfo.CurrentCulture);
+            Assert.Equal(expectedUICultureInfo, CultureInfo.CurrentUICulture);
+            Assert.Equal(ThreadPriority.Normal, Thread.CurrentThread.Priority);
+            Task.Factory.StartNew(() => {
+                if(Interlocked.Increment(ref count) == Environment.ProcessorCount)
+                {
+                    e0.Set();
+                }
+            });
+        }
+        Assert.True(e0.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds));
+    }
+
+    [Fact]
+    public static void GlobalWorkQueueDepletionTest()
+    {
+        ManualResetEvent e0 = new ManualResetEvent(false);
+        int count = 0;
+        int maxCount = Environment.ProcessorCount * 64;
+        object syncRoot = new object();
+        void Job(object _)
+        {
+            if(Interlocked.Increment(ref count) >= maxCount)
+            {
+                e0.Set();
+            }
+            else
+            {
+                ThreadPool.QueueUserWorkItem(Job);
+                ThreadPool.QueueUserWorkItem(Job);
+            }
+        }
+        ThreadPool.QueueUserWorkItem(Job);
+        e0.WaitOne();
+    }
+
+    [Fact]
+    public static void SettingMinThreadsWillCreateThreadsUpToMinimum()
+    {
+        ManualResetEvent e0 = new ManualResetEvent(false);
+        AutoResetEvent jobsQueued = new AutoResetEvent(false);
+        int count = 0;
+        ThreadPool.GetMinThreads(out int minThreads, out int unused);
+        ThreadPool.SetMaxThreads(minThreads, unused);
+        for(int i = 0; i < minThreads + 1; ++i)
+        {
+            ThreadPool.QueueUserWorkItem(_ => {
+                if(Interlocked.Increment(ref count) == minThreads + 1)
+                {
+                    jobsQueued.Set();
+                }
+                e0.WaitOne();
+            });
+        }
+        Assert.False(jobsQueued.WaitOne(ThreadTestHelpers.ExpectedMeasurableTimeoutMilliseconds));
+        Assert.True(ThreadPool.SetMaxThreads(minThreads + 1, unused));
+        Assert.True(ThreadPool.SetMinThreads(minThreads + 1, unused));
+
+        Assert.True(jobsQueued.WaitOne(ThreadTestHelpers.UnexpectedTimeoutMilliseconds));
+
+        e0.Set();
```

Comment:
This test should have a finally block that resets the min and max threads to defaults so that it does not affect other tests in case of failure. More applicable for when we move to using the xunit runner, but would be good to make that change now.